core/modules/contact/contact.admin.inc

```diff
@@ -10,68 +10,90 @@
  *
  * @see contact_menu()
  */
-function contact_category_list() {
+function contact_category_list($form, $form_state) {
+  $form = array();
 
   $config_data = contact_config_data();
-  $selected_category = $config_data['contact_default_category'];
+  $default_category = $config_data['contact_default_category'];
   $categories = $config_data['categories'];
 
-  $header = array(
-    t('Category'),
-    t('Recipients'),
-    t('Selected'),
-    t('Operations'),
-  );
-  $rows = array();
-
+  $form['#tree'] = TRUE;
   // Loop through the categories and add them to the table.
   if ($categories) {
     foreach ($categories as $cat) {
-      $row = array();
-      $row[] = check_plain($cat['category']);
-      $row[] = check_plain($cat['recipients']);
-
-      if ($cat['cid'] == $selected_category) {
-        $row[] = t('Yes');
-      }
-      else {
-        $row[] = t('No');
-      }
+      $cid = $cat['cid'];
 
       $links = array();
       $links['configure'] = array(
         'title' => t('Configure'),
-        'href' => 'admin/structure/contact/configure/' . $cat['cid'],
+        'href' => 'admin/structure/contact/configure/' . $cid,
       );
       $links['delete'] = array(
         'title' => t('Delete'),
-        'href' => 'admin/structure/contact/delete/' . $cat['cid'],
+        'href' => 'admin/structure/contact/delete/' . $cid,
+      );
+
+      $form['categories'][$cid]['weight'] = array(
+        '#type' => 'weight',
+        '#title' => t('Weight for @title', array('@title' => $cat['category'])),
+        '#title_display' => 'invisible',
+        '#default_value' => $cat['weight'],
+        '#delta' => max(10, sizeof($categories)),
+      );
+      $form['categories'][$cid]['name'] = array(
+        '#markup' => check_plain($cat['category']),
+      );
+      $form['categories'][$cid]['recipients'] = array(
+        '#markup' => check_plain($cat['recipients']),
       );
-      $row[] = array(
-        'data' => array(
-          '#type' => 'operations',
-          '#links' => $links,
-        ),
+      $form['categories'][$cid]['default'] = array(
+        '#markup' => ($cid == $default_category) ? t('Yes') : t('No'),
       );
-      $rows[] = $row;
+      $form['categories'][$cid]['operations'] = array(
+        '#type' => 'operations',
+        '#links' => $links,
+      );
+
     }
   }
 
-  if (!$rows) {
-    $rows[] = array(array(
-      'data' => t('No categories available.'),
-      'colspan' => 5,
-    ));
+  $form['actions'] = array('#type' => 'actions', '#tree' => FALSE);
```

Comment:
Rather than setting #tree on the entire form and then setting FALSE on the actions, you could just set $form['categories']['#tree'] = TRUE.